addons/udes_stock/models/stock_move_line.py

```diff
@@ -1120,20 +1120,26 @@ def _prevent_write_after_done(self):
         # When a moveline is updated through desktop only one "done" move line is passed into self even when there are multiple
         # Get all the done move lines on the picking if there is only one "done" move line
         if len(done_lines) == 1 and self.mapped("picking_id"):
-            done_lines = self.mapped("picking_id.move_line_ids").filtered(
-                lambda ml: ml.state == "done"
+            done_lines = self.search(
+                [("picking_id", "in", self.mapped("picking_id.id")), ("state", "=", "done")]
```

Comment:
Couple of minor suggestions:

```
if len(done_lines) == 1:
    picking_ids = self.mapped("picking_id.id")
    if picking_ids:
        done_lines = self.search([("picking_id", "in", self.mapped("picking_id.id")), ("state", "=", "done")], order="id")
```

This avoids us doing `mapped` twice. And `order="id"` means we avoid using the default `_order` on the move line model, as the order of records doesn't matter here.